crates/db_views/src/combined/person_content_combined_view.rs

```diff
@@ -42,11 +43,197 @@ use lemmy_db_schema::{
     community::CommunityFollower,
   },
   traits::InternalToCombinedView,
-  utils::{actions, actions_alias, functions::coalesce, get_conn, DbPool},
+  utils::{functions::coalesce, get_conn, DbPool},
   PersonContentType,
 };
 use lemmy_utils::error::LemmyResult;
 
+impl PersonContentCombinedViewInternal {
+  #[diesel::dsl::auto_type(no_type_alias)]
+  fn joins(my_person_id: Option<PersonId>) -> _ {
+    let item_creator = person::id;
+
+    let comment_join =
+      comment::table.on(person_content_combined::comment_id.eq(comment::id.nullable()));
+
+    let post_join = post::table.on(
+      person_content_combined::post_id
+        .eq(post::id.nullable())
+        .or(comment::post_id.eq(post::id)),
+    );
+
+    let item_creator_join = person::table.on(
+      comment::creator_id
+        .eq(item_creator)
+        // Need to filter out the post rows where the post_id given is null
+        // Otherwise you'll get duped post rows
+        .or(
+          post::creator_id
+            .eq(item_creator)
+            .and(person_content_combined::post_id.is_not_null()),
+        ),
+    );
+
+    let community_join = community::table.on(post::community_id.eq(community::id));
+
+    let creator_community_actions_join = creator_community_actions.on(
+      creator_community_actions
+        .field(community_actions::community_id)
+        .eq(post::community_id)
+        .and(
+          creator_community_actions
+            .field(community_actions::person_id)
+            .eq(item_creator),
+        ),
+    );
+
+    let local_user_join = local_user::table.on(
+      item_creator
+        .eq(local_user::person_id)
+        .and(local_user::admin.eq(true)),
+    );
+
+    let community_actions_join = community_actions::table.on(
+      community_actions::community_id
+        .eq(post::community_id)
+        .and(community_actions::person_id.nullable().eq(my_person_id)),
+    );
```

Comment:
Joins like this should still use BindIfSome for performance.

https://docs.rs/diesel-bind-if-some/latest/diesel_bind_if_some/struct.BindIfSome.html

Also I just noticed an incorrect QueryId implementation for BindIfSome, which could break queries that don't have another thing (e.g. the query being boxed) that sets HAS_STATIC_QUERY_ID to false. I released version 0.2.0 which fixes this.

Reply:
If that's a universal performance benefit, then it'd be better to add that to back-end diesel, because this isn't a specifically lemmy problem, and it could benefit every nullable query.